### PR TITLE
Feature/stl sharedptr implementation

### PR DESCRIPTION
**Pull request by conlain-k**
Addresses #1493.   

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Run unit tests and ATF tests. Will most likely require extensive testing due to number of files altered.  

### Summary
Converts instances of `Utils::SharedPtr` to `std::shared_ptr`, supported in C++11. Also uses `std::make_shared`. Converts non-standard functionality to use STL equivalents, cleans up lots of shared_ptr allocation issues. 
Should only modify shared pointers in-place with no change in functionality. Internal API should be basically identical.

### Changelog
Massive replacement of sharedPtrs
Conversion to use correct allocation processes in shared_ptr initialization
fix namespacing and remove custom implementations of non-standard functionality

NOTE:
Also modifies `application_manager_impl.h` to use ```subscribed_way_points_apps_list_.find(app->app_id())``` instead of ```subscribed_way_points_apps_list_.find(app)```.

The `subscribed_way_points_apps_list_` is a `std::set<int32_t>`, so indexing into it with a shared_ptr makes no sense. This is either poorly defined behavior or a bug. It may be the case that this is outside the scope of this PR, but the modification is required for the code to compile and the current code still passes all unit and SDL smoke tests.

##### Enhancements
* STL-provided pointers are cleaner and more maintainable

##### Bug Fixes
* see above note about the indexing

### Tasks Remaining:
- [x] test aggressively using existing tests

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by conlain-k** [comment]
I added a commit that changed some strange naming wherein TSharedPtr got converted to Tstd::shared_ptr, resulting in an unintended extra namespacing. The change is cosmetic but improves code cleanness.

File: src/components/include/utils/shared_ptr.h [hunk 1]
```diff
@@ -36,353 +36,18 @@
...
-
-template <typename ObjectType>
-inline bool SharedPtr<ObjectType>::valid() const {
-  if (mReferenceCounter && (0 < *mReferenceCounter)) {
-    return (mObject != NULL);
-  }
-  return false;
+/* Test if the given shared ptr is valid and in use
+ */
+template <typename T>
```
Comment (jacobkeeler) [inline src/components/include/utils/shared_ptr.h]: Might do well with a rename of this file, with this being the only function in the file

File: src/components/media_manager/src/media_manager_impl.cc [hunk 2]
```diff
@@ -30,28 +30,28 @@
...
-#include "media_manager/video/pipe_video_streamer_adapter.h"
-#include "media_manager/audio/pipe_audio_streamer_adapter.h"
-#include "media_manager/video/file_video_streamer_adapter.h"
 #include "media_manager/audio/file_audio_streamer_adapter.h"
+#include "media_manager/audio/pipe_audio_streamer_adapter.h"
+#include "media_manager/audio/socket_audio_streamer_adapter.h"
 #include "media_manager/media_manager_settings.h"
+#include "media_manager/video/file_video_streamer_adapter.h"
+#include "media_manager/video/pipe_video_streamer_adapter.h"
+#include "media_manager/video/socket_video_streamer_adapter.h"
```
Comment (jacobkeeler) [inline src/components/media_manager/src/media_manager_impl.cc]: Unrelated

File: src/components/media_manager/src/media_manager_impl.cc [hunk 3]
```diff
@@ -213,9 +218,9 @@ void MediaManagerImpl::StartMicrophoneRecording(int32_t application_key,
   std::vector<uint8_t> buf;
   if (file_system::ReadBinaryFile(record_file_source, buf)) {
     if (file_system::Write(file_path, buf)) {
-      LOG4CXX_INFO(logger_,
-                   "File " << record_file_source << " copied to "
-                           << output_file);
+      LOG4CXX_INFO(
+          logger_,
+          "File " << record_file_source << " copied to " << output_file);
```
Comment (jacobkeeler) [inline src/components/media_manager/src/media_manager_impl.cc]: These changes might conflict with the version of `clang-format` that is currently used by the project.

File: src/components/policy/policy_external/doc/doxygen/components/SmartObjects/Smart Objects Validation/Schema Structure.txt [hunk 4]
```diff
@@ -57,17 +57,17 @@ Example:
 //
 // true, if successful
 // false, if failed
-TSharedPtr<ISchemaItem> success_SchemaItem = CBoolSchemaItem::create(TSchemaItemParameter<bool>());
+Tshared_ptr<ISchemaItem> success_SchemaItem = CBoolSchemaItem::create(TSchemaItemParameter<bool>());
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/doc/doxygen/components/SmartObjects/Smart Objects Validation/Schema Structure.txt]: Looks like this might be a find/replace error

File: src/components/policy/policy_regular/src/policy_manager_impl.cc [hunk 5]
```diff
@@ -44,12 +44,12 @@
...
 #include "utils/logger.h"
 #include "utils/date_time.h"
-#include "utils/make_shared.h"
+#include <memory>
 #include "policy/cache_manager.h"
 #include "policy/update_status_manager.h"
 #include "config_profile/profile.h"
 #include "utils/timer_task_impl.h"
-#include "utils/make_shared.h"
+#include <memory>
```
Comment (jacobkeeler) [inline src/components/policy/policy_regular/src/policy_manager_impl.cc]: Another duplicate include

File: src/components/protocol_handler/src/multiframe_builder.cc [hunk 6]
```diff
@@ -35,7 +35,7 @@
 #include <limits>
 
 #include "utils/logger.h"
-#include "utils/make_shared.h"
+#include <memory>
```
Comment (jacobkeeler) [inline src/components/protocol_handler/src/multiframe_builder.cc]: Can likely be removed

File: src/components/transport_manager/src/transport_manager_impl.cc [hunk 7]
```diff
@@ -1066,7 +1066,7 @@ void TransportManagerImpl::Handle(TransportAdapterEvent event) {
       LOG4CXX_ERROR(logger_, "Transport adapter failed to send data");
       // TODO(YK): potential error case -> thread unsafe
       // update of message content
-      if (event.event_data.valid()) {
+      if (event.event_data && event.event_data.use_count() > 0) {
```
Comment (jacobkeeler) [inline src/components/transport_manager/src/transport_manager_impl.cc]: Any reason that `ValidSPtr` isn't used here, but is used in the USB adapter?

File: tools/InterfaceGenerator/test/generator/generators/test_expected_jsonrpc.cc [hunk 8]
```diff
@@ -69,7 +69,7 @@ XXX::YYY::ZZZ::Test::Test()
   InitFunctionSchemes(struct_schema_items, function_id_items, message_type_items);
 }
 
-TSharedPtr<ISchemaItem> XXX::YYY::ZZZ::Test::ProvideObjectSchemaItemForStruct(
+Tshared_ptr<ISchemaItem> XXX::YYY::ZZZ::Test::ProvideObjectSchemaItemForStruct(
```
Comment (jacobkeeler) [inline tools/InterfaceGenerator/test/generator/generators/test_expected_jsonrpc.cc]: Is this a find/replace error? I haven't found where it could be defined

File: src/components/application_manager/test/commands/command_request_impl_test.cc [hunk 9]
```diff
@@ -65,7 +65,7 @@ using ::testing::Return;
 using ::testing::SaveArg;
 using ::testing::DoAll;
 
-using ::utils::SharedPtr;
+using ::std::shared_ptr;
```
[the remark this answers is not in the document]
Comment (conlain-k) [reply src/components/application_manager/test/commands/command_request_impl_test.cc]: woops, that was a find-replace that became unneccesary. Will remove

File: src/components/application_manager/test/policy_handler_test.cc [hunk 10]
```diff
@@ -53,13 +53,13 @@
...
 #include "utils/file_system.h"
-#include "utils/make_shared.h"
+#include <memory>
 #include "utils/custom_string.h"
 #include "policy/usage_statistics/counter.h"
 #include "policy/usage_statistics/statistics_manager.h"
 #include "interfaces/MOBILE_API.h"
 #include "policy/mock_policy_settings.h"
-#include "utils/make_shared.h"
+#include <memory>
```
[the remark this answers is not in the document]
Comment (conlain-k) [reply src/components/application_manager/test/policy_handler_test.cc]: Unless someone higher in the include stack pulls this in, it's needed for the shared_ptr definitions

File: src/components/transport_manager/src/transport_manager_impl.cc (hunk 7 above)
Comment (conlain-k) [reply src/components/transport_manager/src/transport_manager_impl.cc]: my bad, that should be ValidSPtr

**Comment by jacobkeeler** [comment]
duplicate of #2391
